### PR TITLE
docs(knowledge-base): create git workshop docs for ccss hack the tunnels

### PR DESCRIPTION
**Pull request by MFarabi619**
See attached issue.

Created a follow-along guide under the knowledge base section for the upcoming git workshop hosted by cuHacking. Attached some sample images.

@JowiAoun Let me know if we can get preview deployments for docs.

image

## Welcome page
image
image

## Setup page
image

## Part 1
image

## Part 2
image

## Part 3
image

TODO: Review coderabbit comments and fill in the small, incomplete parts

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive "Hack the Tunnels" documentation series, including guides on shell commands, Git usage, and setting up development environments.
	- Added a structured welcome guide for participants, featuring key concepts and resources related to Linux, Git, and GitHub.
	- Implemented tabbed interfaces for easy navigation in setup and Git command guides.

- **Documentation**
	- Enhanced metadata for better organization and accessibility of documentation content.
	- Updated guides with detailed instructions on automating tasks, version control, and managing Git repositories.

### CODE REVIEW

**Comment by JowiAoun** [comment]
Hey @MFarabi619, preview deploys are now available for `docs`, `portal`, `eslint-config` on PR [#137](https://github.com/cuhacking/2025/pull/137) and will also be enabled for `website` once that is up and working as expected. Preview deploys will be available to all branches once PR is merged in

**Review by HasithDeAlwis — DISMISSED** [review]
yea, this is fire 🔥

File: apps/docs/src/content/docs/knowledge-base/hack-the-tunnels-2024/setup.mdx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+- Search `VS Code`
+
+You can search and find the commands for all of the packages we've installed so far.
+</Callout>
+
+    </div>
+      </div>
+
+
+### Check Your Setup
```
[the remark this answers is not in the document]
Comment (MFarabi619) [reply apps/docs/src/content/docs/knowledge-base/hack-the-tunnels-2024/setup.mdx:201]: Got it. The homebrew part is for both windows and mac users, that's why I had it in as part of the rest of the setup.

This is meant mostly as a reference for CCSS, they're handling pretty much all of the setup for the attendees.

To alleviate the scrollyness a bit, I switched the order so MacOS shows up first.

image

**Comment by MFarabi619** [comment]
Going to bypass branch protections and 2nd reviewer due to workshop being in two days and so that CCSS can have time to review it.

**Pull request merged**